src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -88,4 +96,30 @@ public S3Client getClientForRegion(Region r) {
                 .serviceConfiguration(S3Configuration.builder().useArnRegionEnabled(true).build())
                 .credentialsProvider(credentialsProvider).region(r).build());
     }
+
+    /**
+     * Set s3 endpoint type.
+     *
+     * @param type s3EndpointType
+     */
+    private void setS3EndpointType(String type) {
+        //Check if system property and device config are consistent
+        //If not consistent, set system property according to device config value
+        if (S3_ENDPOINT_DEFAULT_TYPE.equals(type)) {
+            if (System.getProperty(regionalEndpointKey) != null
+                    && System.getProperty(regionalEndpointKey).equals(regionalEndpointValue)) {
+                logger.atInfo().log("s3 endpoint set to global");
+                System.clearProperty(regionalEndpointKey);
+                clientCache.remove(region);
+            }
+        } else {
+            if (System.getProperty(regionalEndpointKey) == null
+                    || !System.getProperty(regionalEndpointKey).equals(regionalEndpointValue)) {
+                logger.atInfo().log("s3 endpoint set to regional");
+                System.setProperty(regionalEndpointKey, regionalEndpointValue);
+                clientCache.remove(region);
+            }
+        }
+    }
+
```

Comment:
Let's also clean up this function - 
1. first convert the system property to a string,
2. then compare with your if-else